src/creature.cpp

```diff
@@ -2403,7 +2434,7 @@ bodypart_id Creature::select_body_part( Creature *source, int hit_roll ) const
     add_msg_debug( debugmode::DF_CREATURE, "target size = %d", get_size() );
     add_msg_debug( debugmode::DF_CREATURE, "difference = %d", szdif );
 
-    return anatomy_human_anatomy->select_body_part( szdif, hit_roll );
+    return anatomy( get_all_body_parts() ).select_body_part( szdif, hit_roll );
```

Comment:
This change is what's causing your test failures. Before, the method was hardcoded to use human anatomy. Now it's based on the anatomy of the creature it's called on. Since default monsters only have head and torso, the distribution is off and arms or legs will never be chosen.
To fix it you could set the defender anatomy in the test to human anatomy or use some other creature that has human anatomy. (And long term add test cases for different anatomies.)

Reply:
well, i changed it out in favor of using an npc's anatomy but it's still not quite right yet.